blaze/compute/tests/test_sql_compute.py

```diff
@@ -809,14 +823,71 @@ def test_str_upper():
     expected = "SELECT upper(accounts.name) as name FROM accounts"
     assert normalize(result) == normalize(expected)
 
-    
+
 def test_str_lower():
     expr = t.name.str_lower()
     result = str(compute(expr, s, return_type='native'))
     expected = "SELECT lower(accounts.name) as name FROM accounts"
     assert normalize(result) == normalize(expected)
 
 
+@pytest.mark.parametrize("sep", [None, " sep "])
+def test_str_cat(sep):
+    """
+    Need at least two string columns to test str_cat
+    """
+
+    if sep is None:
+        expr = t_str_cat.name.str_cat(t_str_cat.comment)
+        expected = """
+                   SELECT accounts2.name || accounts2.comment
+                   AS anon_1 FROM accounts2
+                   """
+    else:
+        expr = t_str_cat.name.str_cat(t_str_cat.comment, sep=sep)
+        expected = """
+                   SELECT accounts2.name || :name_1 || accounts2.comment
+                   AS anon_1 FROM accounts2
+                   """
+
+    result = str(compute(expr, s_str_cat, return_type='native'))
+    assert normalize(result) == normalize(expected)
+
+
+def test_str_cat_chain():
+    expr = (t_str_cat.name
+            .str_cat(t_str_cat.comment, sep=' -- ')
+            .str_cat(t_str_cat.product, sep=' ++ '))
+    result = str(compute(expr, {t_str_cat: s_str_cat}, return_type='native'))
+    expected = """
+               SELECT accounts2.name || :name_1 || accounts2.comment ||
+               :param_1 || accounts2.product AS anon_1 FROM accounts2
+               """
+    assert normalize(result) == normalize(expected)
+
+
+
+@pytest.mark.xfail(reason="raise exception code needed for refactored StrCat")
+@pytest.mark.parametrize("expr",
+                         [t.name.str_cat(t_str_cat.comment),
+                          t.name.str_cat(t_str_cat.comment.str_cat(t_str_cat.name))])
+def test_str_cat_runtime_exception(expr):
```

Comment:
Are we checking that the str_cat operates on the same table? If not, then we should take this test out - I saw a comment that we need to do this check more consistently throughout other blaze operations so perhaps remove the test?